src/main/java/com/cdancy/bitbucket/rest/features/InsightsApi.java

```diff
@@ -0,0 +1,167 @@
+    @Fallback(BitbucketFallbacks.InsightReportPageOnError.class)
+    @GET
+    InsightReportPage listReports(@PathParam("project") String project,
+                                  @PathParam("repo") String repo,
+                                  @PathParam("commitId") String commitId,
+                                  @Nullable @QueryParam("limit") int limit,
+                                  @Nullable @QueryParam("start") int start);
+
+    @Named("insights:get-report")
+    @Documentation( {"https://docs.atlassian.com/bitbucket-server/rest/6.4.0/bitbucket-code-insights-rest.html#idp7"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/projects/{project}/repos/{repo}/commits/{commitId}/reports/{key}")
+    @Fallback(BitbucketFallbacks.InsightReportOnError.class)
+    @GET
+    InsightReport getReport(@PathParam("project") String project,
+                            @PathParam("repo") String repo,
+                            @PathParam("commitId") String commitId,
+                            @PathParam("key") String key);
+
+    @Named("insights:create-report")
+    @Documentation( {"https://docs.atlassian.com/bitbucket-server/rest/6.4.0/bitbucket-code-insights-rest.html#idp9"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/projects/{project}/repos/{repo}/commits/{commitId}/reports/{key}")
+    @Fallback(BitbucketFallbacks.InsightReportOnError.class)
+    @PUT
+    InsightReport createReport(@PathParam("project") String project,
+                               @PathParam("repo") String repo,
+                               @PathParam("commitId") String commitId,
+                               @PathParam("key") String key,
+                               @BinderParam(BindToJsonPayload.class) CreateInsightReport createInsightReport);
+
+    @Named("insights:delete-report")
+    @Documentation( {"https://docs.atlassian.com/bitbucket-server/rest/6.4.0/bitbucket-code-insights-rest.html#idp8"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/projects/{project}/repos/{repo}/commits/{commitId}/reports/{key}")
+    @Fallback(BitbucketFallbacks.RequestStatusOnError.class)
+    @ResponseParser(RequestStatusParser.class)
+    @DELETE
+    RequestStatus deleteReport(@PathParam("project") String project,
+                               @PathParam("repo") String repo,
+                               @PathParam("commitId") String commitId,
+                               @PathParam("key") String key);
+
+    @Named("insights:delete-annotation")
+    @Documentation( {"https://docs.atlassian.com/bitbucket-server/rest/6.4.0/bitbucket-code-insights-rest.html#idp11"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/projects/{project}/repos/{repo}/commits/{commitId}/reports/{key}/annotations")
+    @Fallback(BitbucketFallbacks.RequestStatusOnError.class)
+    @ResponseParser(RequestStatusParser.class)
+    @DELETE
+    RequestStatus deleteAnnotation(@PathParam("project") String project,
```

Comment:
:warning: Rather than using a lot of String arguments, consider using a container object for those values. (#QR16:124#)-PMD_UseObjectForClearerAPI